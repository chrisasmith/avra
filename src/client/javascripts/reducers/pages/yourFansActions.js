import { api } from '../../services'
import { createRequests, doApiCall, createRequestTypes } from '../actionHelper'

export const GET_HOME_PAGE = createRequestTypes('GET_HOME_PAGE')
export const getHomePage = createRequests(GET_HOME_PAGE)

export const HOME_PAGE_DATA_SUCCESS = 'HOME_PAGE_DATA_SUCCESS'
export function homePageDataSuccess(data) {
  return { type: HOME_PAGE_DATA_SUCCESS, data }
}

export function getHomeData() {
  return doApiCall({
    api: api.getHomeData,
    action: getHomePage,
  })
}
