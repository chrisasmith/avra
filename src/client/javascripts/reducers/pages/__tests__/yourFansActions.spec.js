import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import fetchMock from 'fetch-mock'
import * as actions from '../yourFansActions'

const mockStore = configureMockStore([thunk])

describe('YourFans actions', () => {
  afterEach(() => {
    fetchMock.restore()
  })

  it('should call get home data', () => {
    fetchMock.mock('https://fake/getHomeData.json',
      { body: { data: [] } }
    )

    const store = mockStore({})

    return store.dispatch(actions.getHomeData())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot()
      })
  })
})
