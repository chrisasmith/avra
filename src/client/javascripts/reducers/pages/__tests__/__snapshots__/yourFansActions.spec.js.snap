// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourFans actions should call get home data 1`] = `
Array [
  Object {
    "request": Object {},
    "type": "GET_HOME_PAGE_REQUEST",
  },
  Object {
    "request": Object {},
    "response": Object {
      "data": Array [],
    },
    "type": "GET_HOME_PAGE_SUCCESS",
  },
]
`;
